generator/operation_test.go

```diff
@@ -27,35 +27,11 @@ import (
 func TestUniqueOperationNames(t *testing.T) {
 	doc, err := spec.Load("../fixtures/codegen/todolist.simple.yml")
 	if assert.NoError(t, err) {
-		sp := doc.Spec()
-		sp.Paths.Paths["/tasks"].Post.ID = "saveTask"
-		sp.Paths.Paths["/tasks"].Post.AddExtension("origName", "createTask")
-		sp.Paths.Paths["/tasks/{id}"].Put.ID = "saveTask"
-		sp.Paths.Paths["/tasks/{id}"].Put.AddExtension("origName", "updateTask")
-
-		ops := gatherOperations(doc, nil)
-		assert.Len(t, ops, 4)
-		_, exists := ops["saveTask"]
-		assert.True(t, exists)
-		_, exists = ops["saveTask1"]
-		assert.True(t, exists)
-	}
-}
-
-func TestEmptyOperationNames(t *testing.T) {
-	doc, err := spec.Load("../fixtures/codegen/todolist.simple.yml")
-	if assert.NoError(t, err) {
-		sp := doc.Spec()
-		sp.Paths.Paths["/tasks"].Post.ID = ""
-		sp.Paths.Paths["/tasks"].Post.AddExtension("origName", "createTask")
-		sp.Paths.Paths["/tasks/{id}"].Put.ID = ""
-		sp.Paths.Paths["/tasks/{id}"].Put.AddExtension("origName", "updateTask")
-
```

Comment:
I do like the idea of pretty names, and there is a solution to keeping them.  However, that solution is too big to implement now.  We can certainly provide a CLI flag to exercise the former naming convention, but that will bring back the code regeneration breakage.  In effect, we would tell users that you can use this flag to use your IDs for pretty names, but then you cannot have code regeneration.

If the handler names were protected with a namespace (e.g. an interface), then the regeneration breakage gets solved and people can keep their pretty handler names.  It would really clean up configureAPI() and the generated server code.  I really want this, but it's too big for me to tackle now.